### DISCUSSION TITLE
Filter rows based on foreign key row count

### DISCUSSION THREAD

**Discussion by brydonm** [supabase/supabase#7629] [Questions] [answered]
Hey everyone! I'm trying to filter rows based on a count of a foreign row (hopefully that terminology is right lol)

For explanations sake, I have two tables:

`items`
|id|title|
|---|---|
|1|"hello"|
|2|"hey"|

`links`
|id|item_id|title|
|---|---|---|
|1|1|"world"|

I'm hoping to receive something like this back (notice how `item id 2` is not there) because it has no corresponding `links`:
```json
[
  "title": "hello",
  "links": [
    "id": 1
    "title": "world"
  ]
]
```

This is the query I have:

```js
    supabaseClient
      .from("items")
      .select(
        `
          id,
          title,
          url,
          imageSrc,
          slug,
          category,
          links (
            id,
            title
          )
        `
      )
```

I've tried `.gt("count(links)", 0)` along with many other filter methods but I can't seem to get one that works.

Any help is appreciated, thank you!

**Comment by steve-chavez** [comment, marked as answer]
Hm, so you want to get all `items` that have `links` in it right?

Try doing that with:

```js
    supabaseClient
      .from("items")
      .select(
        `
          id,
          title,
          url,
          imageSrc,
          slug,
          category,
          links!inner (
            id,
            title
          )
        `
      )
```

**Reply by brydonm** [reply]
@steve-chavez to the rescue again. Could you let me know the name of that operator/filter so I can look into it and understand how this works?

Thank you!

**Reply by steve-chavez** [reply]
Sure, docs here: https://supabase.com/docs/reference/javascript/select#filtering-with-inner-joins

**Reply by AmeliaIsCoding** [reply]
@steve-chavez, would you be able to help with the case where the foreign row key count is a specific number, not necessarily just 0 or 1?

**Reply by hoverbird** [reply]
+1 to @AmeliaIsCoding's question— I need to filter on a foreign key count that's a specific number, e.g. using .lt() or .lte() etc.

**Reply by steve-chavez** [reply]
Currently you'll need to use a function or view for this, it's not possible directly. Tracking issue: https://github.com/PostgREST/postgrest/issues/2730

(Sorry for the late reply btw)

**Comment by BaccS** [comment]
But how to do vice versa? How to select records where foreign table has no records?

**Reply by uzarsalan** [reply]
I have the same question

**Reply by steve-chavez** [reply]
You can do:

```js
    supabaseClient
      .from("items")
      .select(
        `
          id,
          title,
          url,
          imageSrc,
          slug,
          category,
          links ()
        `
      )
      .is(`links`, null)
```

For more details see https://postgrest.org/en/stable/references/api/resource_embedding.html#null-filtering-on-embedded-resources

**Comment by uzarsalan** [comment]
up with last question
